content/en/docs/instrumentation/cpp/getting-started.md

```diff
@@ -1,185 +1,393 @@
+find_library(OPENTELEMETRY_COMMON_LIB NAMES libopentelemetry_common.a HINTS "${OPENTELEMETRY_ROOT}/build/sdk/src/common" NO_DEFAULT_PATH)
+find_library(OPENTELEMETRY_TRACE_LIB NAMES libopentelemetry_trace.a HINTS "${OPENTELEMETRY_ROOT}/build/sdk/src/trace" NO_DEFAULT_PATH)
+find_library(OPENTELEMETRY_EXPORTER_LIB NAMES libopentelemetry_exporter_ostream_span.a HINTS "${OPENTELEMETRY_ROOT}/build/exporters/ostream" NO_DEFAULT_PATH)
+find_library(OPENTELEMETRY_RESOURCE_LIB NAMES libopentelemetry_resources.a HINTS "${OPENTELEMETRY_ROOT}/build/sdk/src/resource" NO_DEFAULT_PATH)
+
+if(OPENTELEMETRY_COMMON_LIB AND OPENTELEMETRY_TRACE_LIB AND OPENTELEMETRY_EXPORTER_LIB AND OPENTELEMETRY_RESOURCE_LIB)
+	message(STATUS "Found opentelemetry libraries")
+else()
+	message(SEND_ERROR "Did not find opentelemetry libraries")
+endif()
+
+target_link_libraries(dice-server PRIVATE ${OATPP_LIB} ${OPENTELEMETRY_COMMON_LIB} ${OPENTELEMETRY_TRACE_LIB} ${OPENTELEMETRY_EXPORTER_LIB} ${OPENTELEMETRY_RESOURCE_LIB})
 ```
 
-## Run Application
+**OpenTelemetry tracing configuration:**
+
+Update the `main.cpp` file with the following code to initialize a tracer and to emit spans when the `/rolldice` request handler is called.
+
+```cpp
+#include "oatpp/web/server/HttpConnectionHandler.hpp"
+#include "oatpp/network/Server.hpp"
+#include "oatpp/network/tcp/server/ConnectionProvider.hpp"
+
+#include "opentelemetry/exporters/ostream/span_exporter_factory.h"
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/processor.h"
+#include "opentelemetry/sdk/trace/simple_processor_factory.h"
+#include "opentelemetry/sdk/trace/tracer_provider_factory.h"
+#include "opentelemetry/trace/provider.h"
+
+#include <cstdlib>
+#include <ctime>
+#include <string>
+
+using namespace std;
+namespace trace_api = opentelemetry::trace;
+namespace trace_sdk = opentelemetry::sdk::trace;
+namespace trace_exporter = opentelemetry::exporter::trace;
+
+namespace {
+  void InitTracer() {
+    auto exporter  = trace_exporter::OStreamSpanExporterFactory::Create();
+    auto processor = trace_sdk::SimpleSpanProcessorFactory::Create(std::move(exporter));
+    std::shared_ptr<opentelemetry::trace::TracerProvider> provider =
+      trace_sdk::TracerProviderFactory::Create(std::move(processor));
+    //set the global trace provider 
+    trace_api::Provider::SetTracerProvider(provider);  
+  }
+  void CleanupTracer() {
+    std::shared_ptr<opentelemetry::trace::TracerProvider> none;
+    trace_api::Provider::SetTracerProvider(none);
+  }
+  
+} 
+
+class Handler : public oatpp::web::server::HttpRequestHandler {
+public:
+  shared_ptr<OutgoingResponse> handle(const shared_ptr<IncomingRequest>& request) override {
+    auto tracer = opentelemetry::trace::Provider::GetTracerProvider()->GetTracer("my-app-tracer");
+    auto span = tracer->StartSpan("RollDiceServer");
+    int low = 1;
+    int high = 7;
+    srand((int)time(0));
```

Comment:
nit - Although not related to otel-cpp, `srand` initialized within the handle(), seems inefficient and unnecessary. It should be only initialized only during program bootstrap.

Reply:
@Akhaled19 can you fix this?

Reply:
Good point. I moved it inside main()